src/peripheral/dwt.rs

```diff
@@ -356,12 +356,23 @@ pub struct ComparatorAddressSettings {
     pub access_type: AccessType,
 }
 
+/// Settings for cycle count matching
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub struct CycleCountSettings {
+    /// The cycle count value to compare against.
+    pub compare: u32,
+    /// The cycle count mask value to use.
+    pub mask: u32,
+}
+
 /// The available functions of a DWT comparator.
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 #[non_exhaustive]
 pub enum ComparatorFunction {
     /// Compare accessed memory addresses.
     Address(ComparatorAddressSettings),
+    /// Compare cycle count  & target value.
+    CycleCount(CycleCountSettings),
```

Comment:
Document that this function generates a watchpoint debug event, and what section in the reference manual they can read up on it.

Reply:
Moved that documentation to `EmitOption`